src/commands/run.rs

```diff
@@ -0,0 +1,392 @@
+//! The module that implements the `wasmtime run` command.
+
+use crate::{init_file_per_thread_logger, pick_compilation_strategy, CommonOptions};
+use anyhow::{bail, Context as _, Result};
+use std::{
+    collections::HashMap,
+    ffi::{OsStr, OsString},
+    fmt::Write,
+    fs::File,
+    path::{Component, Path, PathBuf},
+};
+use structopt::{clap::AppSettings, StructOpt};
+use wasi_common::preopen_dir;
+use wasmtime::{Config, Engine, HostRef, Instance, Module, Store};
+use wasmtime_environ::{cache_init, settings, settings::Configurable};
+use wasmtime_interface_types::ModuleData;
+use wasmtime_jit::Features;
+use wasmtime_wasi::{
+    create_wasi_instance, old::snapshot_0::create_wasi_instance as create_wasi_instance_snapshot_0,
+};
+#[cfg(feature = "wasi-c")]
+use wasmtime_wasi_c::instantiate_wasi_c;
+#[cfg(feature = "wasi-c")]
+use wasmtime_wasi_c::instantiate_wasi_c;
+
+fn parse_module(s: &OsStr) -> Result<PathBuf, OsString> {
```

Comment:
The error payload of `Result` in this module seems a bit odd, could these all be `anyhow::Result<T>`?

Reply:
Oh or is this controlled by structopt?

Reply:
I'll test it to see if structopt works with an `anyhow::Result`.

Reply:
Unfortunately it looks like `try_from_os_str` expects an `OsString` for error. `try_from_str` does work, so I've converted them to return `anyhow::Result`, but the module path needs to be`try_from_os_str`.

Reply:
Interesting! Not sure what's going on there but hey w/e works works!